docs/section5/Rule5-21.md

```diff
@@ -0,0 +1,66 @@
+
+# Envelope - Rule 5-21  
+
+**Rule ID:** 5-21  
+**Rule Description:** The vertical fenestration shall be distributed on each face of the building in the same proportion as in the proposed design.  
+**Rule Assertion:** B-RMR total (subsurface.glazed_area+subsurface.opaque_area) = expected value for each zone  
+**Appendix G Section:** Section G3.1-5(c) Building Envelope Modeling Requirements for the Baseline building  
+**Appendix G Section Reference:**  None
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:** None  
+
+**Manual Checks:** None  
+**Evaluation Context:**  Each Data Element  
+**Data Lookup:** None  
+**Function Call:**  
+
+  1. get_building_segment_window_wall_areas()  
+  2. match_data_element()
+  3. get_opaque_surface_type()
+  4. get_surface_conditioning_category()
+
+## Rule Logic:
+
+- Get building segment window wall areas dictionary for B_RMR: `window_wall_areas_dictionary_b = get_building_segment_window_wall_areas(B_RMR)`
+
+- Get building segment window wall areas dictionary for P_RMR: `window_wall_areas_dictionary_p = get_building_segment_window_wall_areas(P_RMR)`
+
+- Get surface conditioning category dictionary for B_RMR: `scc_dictionary_b = get_surface_conditioning_category(B_RMR)`  
+
+- For each building segment in the Baseline model: `For building_segment_b in B_RMR.building.building_segments:`
+
+  - Get total fenestration area for building segment: `total_fenestration_area_b = window_wall_areas_dictionary_b[building_segment_b.id][0]`
+
+  - Get matching building segment in P_RMR: `building_segment_p = match_data_element(P_RMR, BuildingSegments, building_segment_b.id)`
+    - Get total fenestration area for building segment in P_RMR: `total_fenestration_area_p = window_wall_areas_dictionary_p[building_segment_p.id][0]`
+
+  - For each thermal block in building segment: `for thermal_block_b in building_segment_b.thermal_blocks:`
+
+    - For each zone in thermal block: `for zone_b in thermal_block_b.zones:`  
+
+      - For each surface in zone: `for surface_b in zone_b.surfaces:`  
+
+        - Check if surface is above-grade wall and is regulated: `if ( get_opaque_surface_type(surface_b.id) == "ABOVE-GRADE WALL" ) AND ( scc_dictionary_b[surface_b.id] != "UNREGULATED" ):`
+
+          - For each subsurface in surface: `for subsurface_b in surface_b.subsurfaces:`
+
```

Comment:
In the following lines, you are calulating the total fenestration area on each regulated surface in the baselnie and proposed design. I don't think this is necessary. I think from line 46 you can jump right to the rule assertion to check that each subsurface accounts for the same percentage of overall fenestration area in the baseline as in the proposed design. You'd only exclude doors that are less than 50% glass from this check; all other regulated subsurfaces would need to be verified.

Reply:
area of each sub-surface should account for the same % of sub-surface area in the baseline as the proposed.
baseline, should have the same number of sub-surfaces as the proposed, i.e., there should be an alignment in subsurface IDs between B and P.

Reply:
this verification should happen at surface level and not zone level.